### PR TITLE
PLAT-5462 Fix EndlistAppender and add logging

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: lib/Worker.js [hunk 1]
```diff
@@ -134,6 +129,8 @@ module.exports = (function(){
                 // entry won't be created for it until the previous one's creation has completed
                 that.handledEntries[e.entryId] = 'pending';
                 logger.info("[%s] Entry started playing - creating a new object for it", e.entryId);
+                logger.debug("[%s] Removing from endlistAppender", e.entryId);
+                that.endListAppender.removeEntry(e.entryId) // Entry is alive
```
Comment (kalturaguy) [inline lib/Worker.js]: add try catch?

File: lib/manifest/ChunklistManifestGenerator.js [hunk 2]
```diff
@@ -55,6 +55,9 @@ module.exports = function(outputFolderPath, filename, timeWindowSize, maxChunkCo
         {
             // Update in-memory manifest
             currentManifest = manifest;
+            if (manifest.get('playlistType')) {
+                delete manifest.properties['playlistType'];
+            }
             return currentManifest;
```
Comment (kalturaguy) [inline lib/manifest/ChunklistManifestGenerator.js]: please write a comment why we have this code...

File: lib/manifest/EndListAppender.js [hunk 3]
```diff
@@ -98,10 +101,15 @@ var EndlistAppender = function EndListAppender(){
         return qio.write(endListPendingFilePath, JSON.stringify(_.keys(list)));
     };
 
+    var printEndlistAppenderList = function() {
+        logger.debug("Entries to process: %s", JSON.stringify(_.keys(this._entriesToProcess)));
```
Comment (kalturaguy) [inline lib/manifest/EndListAppender.js]: use %j instead of stringfy

File: lib/manifest/ChunklistManifestGenerator.js (hunk 2 above)
Comment (gadyaari) [inline lib/manifest/ChunklistManifestGenerator.js]: done

File: lib/manifest/EndListAppender.js (hunk 3 above)
Comment (gadyaari) [inline lib/manifest/EndListAppender.js]: done

File: lib/Worker.js [hunk 4]
```diff
@@ -84,11 +84,6 @@ module.exports = (function(){
                     logger.debug("[%s] Adding entry to list of handled entries", e.entryId);
                     that.handledEntries[e.entryId] = entry;
                     //entry.on('stopped', onLiveEntryStopped.bind(that));
-                    logger.debug("[%s] Removing entry from endlistAppender", e.entryId);
-                    that.endListAppender.removeEntry(e.entryId) // Entry is alive
-                        .catch(function(err){
-                            logger.error("[%s] Error adding entry to endlist appender processing - %s", e.entryId, err.message);
```
Comment (kalturaguy) [inline lib/Worker.js]: what if err is null? or a string? you will have an unhandled exception...
talk with Ron, he added a new utility to print errors

File: lib/Worker.js [hunk 5]
```diff
@@ -134,6 +129,11 @@ module.exports = (function(){
                 // entry won't be created for it until the previous one's creation has completed
                 that.handledEntries[e.entryId] = 'pending';
                 logger.info("[%s] Entry started playing - creating a new object for it", e.entryId);
+                logger.debug("[%s] Removing from endlistAppender", e.entryId);
+                that.endListAppender.removeEntry(e.entryId)
+                    .catch(function() {
+                        logger.error("Error removing entry to endlistAppender");
```
Comment (kalturaguy) [inline lib/Worker.js]: what about printing the err in the catch....?

File: lib/manifest/ChunklistManifestGenerator.js [hunk 6]
```diff
@@ -55,6 +55,12 @@ module.exports = function(outputFolderPath, filename, timeWindowSize, maxChunkCo
         {
             // Update in-memory manifest
             currentManifest = manifest;
+            // When manifest has playlistType property it will add END-LIST to chunklist.
+            // By removing it when module start function we make sure while entry is updating and live not to have
+            // END-LIST at the end of the file.
+            if (manifest.get('playlistType')) {
+                delete manifest.properties['playlistType'];
```
Comment (ronyad) [inline lib/manifest/ChunklistManifestGenerator.js]: I this shouldnt happen normally, I thing that you should ass warning message

File: lib/manifest/EndListAppender.js [hunk 7]
```diff
@@ -57,12 +58,14 @@ var EndlistAppender = function EndListAppender(){
...
-                }).catch(function(err){
-                    logger.error('Error adding EXT-X-ENDLIST: ' + err.message);
-                });
-            }, sessionManager.getSessionDuration());
+                logger.debug("Timer ended. Starting endlistAppender");
+                appendEndlistIfMissing.call(that, entryId)
+                    .then(function(){
+                        return that.removeEntry(entryId);
+                    }).catch(function(err){
+                        logger.error('Error adding EXT-X-ENDLIST: ' + err.message);
```
Comment (ronyad) [inline lib/manifest/EndListAppender.js]: logger.error('Error adding EXT-X-ENDLIST:', ErrorUtils.error2string(err));
And of course  add 
 var ErrorUtils = require('./../utils/error-utils');

File: lib/manifest/ChunklistManifestGenerator.js (hunk 6 above)
Comment (gadyaari) [inline lib/manifest/ChunklistManifestGenerator.js]: done

File: lib/manifest/EndListAppender.js [hunk 8]
```diff
@@ -57,12 +58,14 @@ var EndlistAppender = function EndListAppender(){
                 return;
             }
             var timeoutObj = setTimeout(function () {
-                appendEndlistIfMissing.call(that, entryId).then(function(){
-                    return that.removeEntry(entryId);
-                }).catch(function(err){
-                    logger.error('Error adding EXT-X-ENDLIST: ' + err.message);
-                });
-            }, sessionManager.getSessionDuration());
+                logger.debug("Timer ended. Starting endlistAppender");
```
Comment (ronyad) [inline lib/manifest/EndListAppender.js]: Add the entryid to all logging.

File: lib/Worker.js [hunk 9]
```diff
@@ -134,6 +129,11 @@ module.exports = (function(){
...
+                    });
```
Comment (ronyad) [inline lib/Worker.js]: Note that if createLiveEntry is failed, then entry can still be "offline" but without end-list

**Pull request merged**